x-pack/plugins/stack_connectors/server/connector_types/openai/lib/azure_openai_utils.test.ts

```diff
@@ -171,26 +171,44 @@ describe('Azure Open AI Utils', () => {
 
     it('should not match invalid chat url', () => {
       const regex = transformApiUrlToRegex(AZURE_OPENAI_CHAT_URL);
-      const match =
-        'https://openai.azure.com/openai/deployments/NEW-DEPLOYMENT-321/chat/completions?api-version=2023-05-15'.match(
-          regex
-        );
-      expect(match).toBeNull();
+      [
+        'https://openai.azure.com/openai/deployments/NEW-DEPLOYMENT-321/chat/completions?api-version=2023-05-15',
+        'https://My-test-resource-123.openaiazure.com/openai/deployments/NEW-DEPLOYMENT-321/chat/completions?api-version=2023-05-15',
```

Comment:
__optional nit:__ I'd also recommend adding a case where a dot is replaced with any symbol to make sure we properly escape dots specifically in the resulting regular expression. So in addition to `openaiazure.com` (omitted `.`), also test `openaiaazure.com` (using `a` symbol instead of `.`) or something along these lines.

Reply:
Updated in https://github.com/elastic/kibana/pull/177635/commits/97980c904bd98d657ed05ac44dbee84d2381194d